src/browser/api/system.js

```diff
@@ -220,25 +221,50 @@ module.exports.System = {
             errorCallback('Failed to send a message to the RVM.');
         }
     },
-    exit: function( /*callback*/ ) {
+    exit: function() {
         electronApp.quit();
     },
-    getAllWindows: function( /*callback, errorCallback*/ ) {
+    getAllWindows: function() {
         return coreState.getAllWindows();
     },
-    getAllApplications: function( /*callback , errorCallback*/ ) {
+    getAllApplications: function() {
         return coreState.getAllApplications();
     },
-    getCommandLineArguments: function( /*callback, errorCallback*/ ) {
+    getCommandLineArguments: function() {
         return electronApp.getCommandLineArguments();
     },
-    getConfig: function( /*callback, errorCallback*/ ) {
+    getConfig: function() {
         return coreState.getStartManifest();
     },
-    getDeviceId: function( /*callback, errorCallback*/ ) {
+    getDeviceUserId: function() {
+        const hash = crypto.createHash('sha256');
+
+        let hostToken;
+        let username;
+
+        if (process.platform === 'darwin') {
+            hostToken = os.networkInterfaces().en0[0].mac;
+            username = process.env.USER;
+        } else {
+
+            // assume windows
+            hostToken = electronApp.getHostToken();
+            username = process.env.USERNAME;
+        }
+
+        if (!username || !hostToken) {
+            throw new Error(`One of username (${username}) or host token (${hostToken}) not defined `);
```

Comment:
space yes, but I actually like throwing here, this is an API call where throwing will trigger the error callback for the caller